### ISSUE TITLE
Jackson date format doesn't respect system time zone since 1.3.0.M3

### ISSUE THREAD

**Issue by izeye** [spring-projects/spring-boot#4070]
Before 1.3.0.M3, Jackson date format (via `spring.jackson.date-format`) follows system time zone but after 1.3.0.M3 (including 1.3.0.BUILD-SNAPSHOT), it doesn't follow system time zone.

It could be worked around by downgrading Jackson to 2.5.4 via `ext['jackson.version'] = '2.5.4'` or setting time zone explicitly via `spring.jackson.time-zone=GMT+9`.

This is a sample project reproducing the situation:

https://github.com/izeye/spring-boot-throwaway-branches/tree/rest-and-jackson

**Comment by wilkinsona** [comment]
This appears to be a bug/change in behaviour in Jackson Databind. It can be reproduced without involving Spring Boot:

```java
ObjectMapper objectMapper = new ObjectMapper();
objectMapper.setDateFormat(new SimpleDateFormat("yyyy-MM-dd HH:mm:ss"));
System.out.println(objectMapper.writeValueAsString(new Date()));
```

Dropping back to 2.6.0 also fixes the problem, i.e. the change appears to have been introduced in 2.6.1. I've opened [a `jackson-databind` issue](https://github.com/FasterXML/jackson-databind/issues/954).

I'm going to close this issue. We'll upgrade to new Jackson 2.6.x versions as a matter of course and you seem to have a good workaround for now (setting `spring.jackson.time-zone`).

**Comment by wilkinsona** [comment]
It looks like this will be fixed in Jackson 2.6.3

**Comment by izeye** [comment]
@wilkinsona Thanks for the comments!
